src/compiler/diagnosticMessages.json

```diff
@@ -1405,7 +1405,7 @@
         "category": "Error",
         "code": 2474
     },
-    "'const' enums can only be used in property or index access expressions or the right hand side of an import declaration or export assignment.": {
+    "'const' enum can only be used in property or index access expressions, the right hand side of an import declaration or export assignment, a typeof query.": {

```

Comment:
Similar to @DanielRosenwasser's idea:
`"A 'const' enum is only allowed in a property access, import declaration, export assignment, or type query."`

Many of the details in the original message, while true, are extraneous.